cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,103 @@
+        Navigatable.__init__(self, appliance)
         self.name = name
         self.description = description
         self.items = items
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
-        web_ui.fill(form, {'name_text': self.name,
-                           'description_text': self.description,
-                           'button_multiselect': self.items},
-                    action=form.add_button)
-        flash_str = 'Catalog "{}" was saved'
-        flash.assert_success_message(flash_str.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name': self.name,
+                   'description': self.description})
+        view.add_button.click()
+        view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        web_ui.fill(form, {'name_text': updates.get('name', None),
-                           'description_text': updates.get('description', None),
-                           'button_multiselect': updates.get('items', None)},
-                    action=form.save_button)
-        flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
-
-    def delete(self, from_dest='All'):
-        """
-        Delete the catalog, starting from the destination provided by from_dest
-        Throws cfme.DestinationNotFound exception
-
-        :param from_dest: A valid navigation destination to start the delete from
-        :return: none
-        """
-        if from_dest in navigator.list_destinations(self):
-            navigate_to(self, from_dest)
-        else:
-            msg = 'cfme.services.catalogs.catalog does not have destination {}'.format(from_dest)
-            raise DestinationNotFound(msg)
-
-        # Delete using the appropriate method
-        if from_dest == 'All':
-            # Select the row to delete, assuming default List View for All
-            listview_table.select_row_by_cells({'Name': self.name, 'Description': self.description})
-            cfg_btn(version.pick({version.LOWEST: 'Remove Items from the VMDB',
-                    '5.7': 'Remove Catalogs'}), invokes_alert=True)
-        elif from_dest == 'Details':
-            cfg_btn(version.pick({version.LOWEST: 'Remove Item from the VMDB',
-                    '5.7': 'Remove Catalog'}), invokes_alert=True)
-
-        sel.handle_alert()
-        flash.assert_success_message(
+        view = navigate_to(self, 'Edit')
+        view.fill({'name': updates.get('name', None),
+                   'description': updates.get('description', None)})
+        view.save_button.click()
+        view.flash.assert_success_message('Catalog "{}" was saved'.format(self.name))
```

Comment:
Check this out: https://github.com/ManageIQ/integration_tests/blob/master/cfme/automate/explorer/namespace.py#L217

This is how I rewrote the CRUD methods for use with widgetastic.